src/Pages/ProductListPage.js

```diff
@@ -1,93 +1,61 @@
 
-import productsMock from 'mocks/en-us/products.json';
-
-const allProducts = productsMock.results;
 
 const ProductListPage = () => {
-    const { data:Categories, isLoading:areCategoriesLoaded } = useFeaturedBanners('category', 30);
-
-    const [ chosedCategories, setChosedCategories ] = useState({ items: [] });
-    const [ filteredProducts, setFilteredProducts ] = useState({ products: [] });
-    const [ hasTimePassed, setHasTimePassed ] = useState(false);
-    const [ showResetButton, setShowResetButton] = useState(false);
     
-    const getTheFilteredProducts = () => {
-        return allProducts.filter( (item) => 
-            chosedCategories.items.indexOf(item.data.category.slug) > -1
-        );
-    };
-
-
-    const addCategory = (array, name) =>{
-        array.push(name);
-
-        setChosedCategories({
-            items: array
-        });
-    };
-
-
-    const removeCategory = (array, name) =>{
-        const indexToRemove = array.findIndex(element => element === name);
-        array.splice(indexToRemove,1);
-
-        setChosedCategories({
-            items: array
-        });
-    };
-
-
-    const handleCheckboxChange = (event) => {
-        const arrayOfchosedCategories = chosedCategories.items;
-
-        const isItOnTheArray = arrayOfchosedCategories.some(
-            category => category === event.target.name
-        );
-
-        if(!isItOnTheArray && event.target.checked)
-            addCategory(arrayOfchosedCategories, event.target.name);
-        if(isItOnTheArray && !event.target.checked)
-            removeCategory(arrayOfchosedCategories, event.target.name);
-
-        if(chosedCategories.items.length > 0)
-            setShowResetButton(true)
-
-        const resultsFromTheFilter = getTheFilteredProducts();
-
-        setFilteredProducts({
-            products: resultsFromTheFilter
-        });
-    };
-
-    const resetFilters = () => {
-        setFilteredProducts({ products: [] });
-        setChosedCategories({ items: [] });
-    }
-
-    //If no category is selected show all products
-    useEffect(() => {
-
-        if(filteredProducts.products.length === 0){
-            setFilteredProducts({ products: allProducts });
-            setShowResetButton(false);
+    const {
+        pager,
+        setPager,
+        numberPerPage,
+        numberOfPages,
+        nextPage,
+        prevPage,
+        currentPage
+    } = usePager();
+    
+    const {
+        handleCheckboxChange,
+        resetFilters,
+        setFilteredProducts,
+        showResetButton,
+        filteredProducts,
+        Categories,
+        areCategoriesLoaded,
+        allProducts,
+        areProductsLoaded
+    } = useCategoriesFilter();
+
+    useEffect(()=>{
+        if(!areProductsLoaded){
+            setPager({
+                ...pager,
+                numberOfItems: allProducts.length,
+                numberOfPages: Math.ceil(allProducts.length/numberPerPage)
+            })            
```

Comment:
Maybe try moving this inside your custom hook and call it when needed. Modifying the state directly may cause you hard-to-debug problems later.